### PR TITLE
ENG-7831 RN options param on a new builder

### PR DESCRIPTION
**Pull request by kishisaka-nid**
ReactNative option handling in the configure() method has been added to the Android SDK to get the SDK to parity with iOS. The `environment` option allows switching between production and develop. The `isAdvancedDevice` allows enabling the advanced device feature. 

To test this, either build the ReactNative libs using ./gradlew NeuroID:assemble, copy and configure the ReactNative SDK to use the SDK libs or use the snapshots of this branch from jitpack. Pull the ENG-7831 branch in the ReactNative SDK repo, yarn and run NPX in the ReactNative SDK sandbox and modify App.tsx in the sandbox app with the following code to add the new remote options: 

`await NeuroID.configure('key_live_MwC5DQNYzRsRhnnYjvz1fJtp',  {
      environment:'prodscriptdevcollection',isAdvancedDevice:true,usingReactNavigation:true,
    });`

If it all works you should see this debug in logcat on running the ReactNative sandbox app with the appropriate configuration that was set in the sandbox app. 
`2024-07-10 23:07:50.680 20069-20131 NIDRNBuilder            com...idreactnativesdksandbox.debug  D  NIDRNBuilder isAdvancedDevice true environment: prodscriptdevcollection`

### CODE REVIEW

File: NeuroID/src/test/java/com/neuroid/tracker/callbacks/ActivityCallbacksUnitTests.kt [hunk 1]
```diff
@@ -218,20 +219,21 @@ internal class ActivityCallbacksUnitTests {
...
-                mocks.mockedActivity,
-                true,
-                true,
-                "activity",
-                parent = expectedActivityName,
-            )
-
-            mocks.mockedRegistration.registerWindowListeners(mocks.mockedActivity)
+        if (!BuildConfig.FLAVOR.contains("react")) {
+            verify(exactly = 1) {
```
Comment (kishisaka-nid) [inline NeuroID/src/test/java/com/neuroid/tracker/callbacks/ActivityCallbacksUnitTests.kt:223]: in the react native version, this happens externally in the app.

File: NeuroID/src/reactNative/java/com/neuroid/tracker/extensions/NIDRNBuilder.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            if (rnOptionsMap.hasKey(RNConfigOptions.isAdvancedDevice.name)) {
+                rnOptionsMap.getBoolean(RNConfigOptions.isAdvancedDevice.name).let {
+                    isAdvancedDevice = it
+                }
+            }
+            // set the environment params from the environment option, default PRODUCTION
+            if (rnOptionsMap.hasKey(RNConfigOptions.environment.name)) {
+                rnOptionsMap.getString(RNConfigOptions.environment.name)?.let {
+                    when (it) {
+                        NeuroID.PRODUCTION -> environment = NeuroID.PRODUCTION
```
Comment (kevinsites) [inline NeuroID/src/reactNative/java/com/neuroid/tracker/extensions/NIDRNBuilder.kt:43]: NIT: Do we need this check if the default catch will set it to prod?

**Review by kevinsites — APPROVED** [review]

File: NeuroID/src/reactNative/java/com/neuroid/tracker/extensions/NIDRNBuilder.kt [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+class NIDRNBuilder( val application: Application? = null,
+                    val clientKey: String = "",
+                    private val rnOptions: ReadableMap? = null) {
+    fun build() {
+        val options = parseOptions(rnOptions)
+        // call NeuroID builder to initialize NeuroID with the options.
+        Log.d("NIDRNBuilder", "set options: $options")
+        NeuroID.Builder(application, clientKey,
+            options[RNConfigOptions.isAdvancedDevice] as Boolean,
+            options[RNConfigOptions.environment] as String).build()
```
Comment (kevinsites) [inline NeuroID/src/reactNative/java/com/neuroid/tracker/extensions/NIDRNBuilder.kt:18]: Could we move the setIsRN() method to here so we are consistent with iOS?

**Review by kevinsites — CHANGES_REQUESTED** [review]

File: NeuroID/src/reactNative/java/com/neuroid/tracker/extensions/NIDRNBuilder.kt (hunk 3 above)
Comment (kishisaka-nid) [reply NeuroID/src/reactNative/java/com/neuroid/tracker/extensions/NIDRNBuilder.kt:18]: yeah I can do that.

Comment (kishisaka-nid) [reply NeuroID/src/reactNative/java/com/neuroid/tracker/extensions/NIDRNBuilder.kt:18]: updated

Comment (kishisaka-nid) [reply NeuroID/src/reactNative/java/com/neuroid/tracker/extensions/NIDRNBuilder.kt:18]: on iOS it looks like setIsRN() is in both places. you want it both places like iOS?

**Review by kevinsites — APPROVED** [review]

**Review by priya-xavier — APPROVED** [review]

**Pull request merged**